src/planscape/eval/compute_conditions.py

```diff
@@ -63,23 +70,48 @@ def read(self, filepath: str, condition_type: ConditionScoreType) -> Optional[Co
             case ConditionScoreType.TRANSFORM:
                 file = 'transform.tif'
         with rasterio.open(os.path.join(self._root_directory, filepath) + file) as src:
-            return src.read(1, out_shape=(1, int(src.height), int(src.width)))
+            return RasterData(src.read(1, out_shape=(1, int(src.height), int(src.width))), src.profile)
+
 
+def _summarize(no_data_value: float, input: list[Optional[RasterData]], operation: str) -> Optional[RasterData]:
+    """Returns a computed raster based on the given operation and inputs and a corresponding profile. The profile of the first input is returned. Profiles are assumed to be consistent across inputs."""
+    if not input:
+        return None
+
+    if any(raster_data is None for raster_data in input):
+        return None
+
+    conditions = [raster_data.raster.astype(
+        'float32') for raster_data in input if raster_data is not None]
+    profiles = [
+        raster_data.profile for raster_data in input if raster_data is not None]
 
-def _summarize(no_data_value: float, input: list[Optional[ConditionMatrix]], operation: str) -> Optional[ConditionMatrix]:
-    conditions = [condition for condition in input if condition is not None]
     output = None
     if conditions:
         if len(conditions) == 1:
             output = conditions[0]
+            output_is_nodata = np.isnan(output) if np.isnan(
+                no_data_value) else (output == no_data_value)
+            output[output_is_nodata] = np.nan
         elif operation == 'MIN':
             output = functools.reduce(np.minimum, conditions)
         else:  # MEAN
             output = average_condition(no_data_value, conditions)
-    return output
+    return RasterData(output, profiles[0])
+
 
+def convert_metric_nodata_to_nan(condition_reader: ConditionReader, metric: Metric, condition_type: ConditionScoreType, is_raw: bool = False) -> Optional[RasterData]:
+    """Replaces NoData values in the raster with NaN. Returns an profile with the updated NoData value."""
+    if not 'filepath' in metric:
+        return None
+    condition = condition_reader.read(
+        metric['filepath'], condition_type, is_raw) if metric['filepath'] else None
+    new_profile = condition.profile
+    new_profile['nodata'] = np.nan
+    return RasterData(convert_nodata_to_nan(condition.profile['nodata'], condition.raster), new_profile)
```

Comment:
Having trouble resolving the types: new_profile seems to be a dictionary, but RasterData seems to want a str for its second argument. Can you clarify?